### PR TITLE
Strip further unused options from micro_p3.F90

### PR DESCRIPTION
**Pull request by PeterCaldwell**
Hardcoded vertical index=1 is top-of-atmosphere in order to get rid of 'model' input variable. Removed optional outputs for precipitation species and user-defined stuff: we can re-add them in a more E3SM-centric way later if we need to.

### CODE REVIEW

**Comment by PeterCaldwell** [comment]
Hey all - I just deleted stuff and regression-tested in this PR. Should I just integrate it without a buddy check, or should I assign someone to review? Thanks...

File: components/scream/src/physics/p3/p3_f90.cpp [hunk 1]
```diff
@@ -76,7 +62,7 @@ FortranDataIterator::FortranDataIterator (const FortranData::Ptr& d) {
 
 void FortranDataIterator::init (const FortranData::Ptr& dp) {
   d_ = dp;
-  fields_.reserve(33);
+  fields_.reserve(23);
```
Comment (ambrad) [inline components/scream/src/physics/p3/p3_f90.cpp]: If you find this line annoying, you can remove it. It was a silly microoptimization.

Comment (ambrad) [reply components/scream/src/physics/p3/p3_f90.cpp]: (On my part, I mean.)

**Comment by ambrad** [comment]
I reviewed since I saw your comment, but feel free to merge w/o review when you judge that to be the way to go. (Same goes for all of us, IMO.)

File: components/scream/src/physics/p3/p3_f90.cpp (hunk 1 above)
Comment (PeterCaldwell) [reply components/scream/src/physics/p3/p3_f90.cpp]: Thanks. If I understand correctly, the .reserve(x) just tells the compiler that you will need enough space for x elements in your vector, right? Thanks for the review.

Comment (ambrad) [reply components/scream/src/physics/p3/p3_f90.cpp]: It acts at runtime. It makes the capacity of the vector at least that big. That means there won't be reallocs behind the scenes until the capacity is exceeded. But this is just test code, so it's a silly micro-opt.

**Pull request merged**